dbms/src/Flash/Coprocessor/GenSchemaAndColumn.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <Flash/Coprocessor/GenSchemaAndColumn.h>
+#include <Storages/Transaction/TypeMapping.h>
+#include <fmt/core.h>
+namespace DB
+{
+DAGSchema genSchema(const TiDBTableScan & table_scan)
+{
+    DAGSchema schema;
+    schema.reserve(table_scan.getColumnSize());
+    for (Int32 i = 0; i < table_scan.getColumnSize(); ++i)
+    {
+        String name = fmt::format("mock_table_scan_{}", i);
+        TiDB::ColumnInfo column_info;
+        auto const & ci = table_scan.getColumns()[i];
+        column_info.tp = static_cast<TiDB::TP>(ci.tp());
+        column_info.id = ci.column_id();
+        column_info.name = name;
+        schema.push_back({name, column_info});
+    }
+    return schema;
+}
+
+NamesAndTypes genNamesAndTypes(const TiDBTableScan & table_scan)
+{
+    NamesAndTypes names_and_types;
+    names_and_types.reserve(table_scan.getColumnSize());
+
+    for (Int32 i = 0; i < table_scan.getColumnSize(); ++i)
+    {
+        String name = fmt::format("mock_table_scan_{}", i);
+        TiDB::ColumnInfo column_info;
+        auto const & ci = table_scan.getColumns()[i];
+        column_info.tp = static_cast<TiDB::TP>(ci.tp());
+        column_info.id = ci.column_id();
+        auto type = getDataTypeByColumnInfoForComputingLayer(column_info);
+        names_and_types.push_back({name, type});
```

Comment:
`TiDBPkColumnID`, `ExtraTableIDColumnID`, `ExtraTableIDColumnID` need to be considered.

like
https://github.com/pingcap/tiflash/blob/dec116f1e9c0caa870b704fad650a5dc1469d6ef/dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp#L603-L655